src/audio/sound_buffer.rs

```diff
@@ -101,8 +101,9 @@ impl SoundBufferRef {
     }
 
     /// Get the samples stored in the buffer
-    pub fn get_samples(&self) -> &[i16] {
-        unsafe { slice::from_raw_parts(ffi::sfSoundBuffer_getSamples(self.raw()), self.sample_count() as usize) }
+    pub fn samples(&self) -> &[i16] {
+        let len = self.sample_count().max(0).min(usize::max_value() as i64) as usize;
```

Comment:
Could you rebase this on top of master? I added a commit that fixes the signature of sample_count to be `u64` instead of the incorrect `i64`.

I also think it would be more appropriate to panic than silently clamp the value.